src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -1530,7 +1545,7 @@ void QgsAdvancedDigitizingDockWidget::CadConstraint::setRelative( bool relative
 void QgsAdvancedDigitizingDockWidget::CadConstraint::setValue( double value, bool updateWidget )
 {
   mValue = value;
-  if ( updateWidget )
+  if ( updateWidget && mLineEdit->isEnabled() )
     mLineEdit->setText( QLocale().toString( value, 'f', 6 ) );
```

Comment:
6 is maybe too much for cartesian, but one could want more digits (7 / 8) for geographic